core/src/test/java/uk/gov/gchq/koryphe/impl/function/ToSetTest.java

```diff
@@ -97,6 +156,77 @@ public void shouldReturnAGivenSet() {
         assertSame(value, result);
     }
 
+    @Test
+    public void shouldReturnAGivenTreeSet() {
+        // Given
+        final ToSet function = new ToSet(TreeSet.class);
+        final Object value = new TreeSet(Sets.newHashSet("value1", "value2"));
+
+        // When
+        final Object result = function.apply(value);
+
+        // Then
+        assertSame(value, result);
+    }
+
+    @Test
+    public void shouldConvertTreeSetToHashSet() {
+        // Given
+        final ToSet function = new ToSet(HashSet.class);
+        final Set expected = Sets.newHashSet("value1", "value2");
+        final Set value = new TreeSet(expected);
+
+        // When
+        final Object result = function.apply(value);
+
+        // Then
+        assertEquals(expected.getClass(), result.getClass());
+        assertNotEquals(value.getClass(), result.getClass());
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void shouldConvertHashSetToTreeSet() {
+        // Given
+        final ToSet function = new ToSet(TreeSet.class);
+        final Set value = Sets.newHashSet("value1", "value2");
+        final Set expected = new TreeSet(value);
+
+
+        // When
+        final Object result = function.apply(value);
+
+        // Then
+        assertEquals(expected.getClass(), result.getClass());
+        assertNotEquals(value.getClass(), result.getClass());
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void shouldHaveHashSetAsDefault() {
+        // Given
+        final ToSet function = new ToSet();
+        final Object value = "value1";
+
+        // When
+        final Object result = function.apply(value);
+
+        // Then
+        assertEquals(HashSet.class, function.getImplementation());
+        assertEquals(HashSet.class, result.getClass());
+        assertEquals(Sets.newHashSet(value), result);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenUnrecognisedSetImplementationUsed() {
+        // Given
+        final ToSet function = new ToSet(LinkedHashSet.class);
+
+        // When / Then
+        IllegalArgumentException e = assertThrows(IllegalArgumentException.class, () -> function.apply(null));
+        assertEquals("Unrecognised Set implementation", e.getMessage());
+    }
+
     @Override
```

Comment:
You could add the implementation field to `shouldJsonSerialiseAndDeserialise` test to make sure it works as expected